server/src/main/java/org/elasticsearch/common/lucene/Lucene.java

```diff
@@ -398,6 +398,8 @@ public static FieldDoc readFieldDoc(StreamInput in) throws IOException {
             return in.readBoolean();
         } else if (type == 9) {
             return in.readBytesRef();
+        } else if (type == 10) {
+            return new BigInteger(in.readString());
```

Comment:
I'm a bit nervous that we didn't have any test failures because this was missed - do we have any tests that check sorting via BigInteger?

Reply:
@romseygeek  Thank you for the review.  We do have tests that sort on BigInteger: `UnsignedLongTests`, and yml test: `unsigned_long/10_basic.yml`.  The serialization of SortValues is used in very particular circumstances (when bottomSortValue is involved, with collapseSort and may that's all); I could not come up with easy examples,  I will try to do this for a following PR.

A usual distributed search on  BigInteger sort doesn't involve serialization of `SortValues`. `SearchHit` arrays come from shards with empty `sortValues`, and  `sortValues` [get filled](https://github.com/elastic/elasticsearch/blob/master/server/src/main/java/org/elasticsearch/action/search/SearchPhaseController.java#L352)  on a coordinating node in the last stage before presenting results to a user.

Reply:
That sounds fine, thanks @mayya-sharipova